packages/maskbook/src/plugins/Trader/trending/useCoinLatestPrice.ts

```diff
@@ -0,0 +1,47 @@
+import { PluginTraderRPC } from '../messages'
+import { useEffect, useState } from 'react'
+import { DataProvider } from '../types'
+import { useValueRef } from '../../../utils/hooks/useValueRef'
+import { subscribeCoins } from '../settings'
+
+export function useCoinLatestPriceInfo(coinId: string | undefined, dataProvider?: DataProvider) {
+    const [latestPriceInfo, setLatestPriceInfo] = useState<{
+        latestPrice: number | null
+        latestPrice_1h_percentage: number | null
+    }>({
+        latestPrice: null,
+        latestPrice_1h_percentage: null,
+    })
+
+    const coins = useValueRef(subscribeCoins)
+
+    // @ts-ignore
```

Comment:
Why?

Reply:
There is a `Not all code paths return a value` error, but I don't actually need return

Reply:
You can return an empty function.  ts-ignore is too powerful and causes a refactoring hazard.